src/features/DisplayControls/DrugChart/components/DrugChartView.jsx

```diff
@@ -188,22 +171,32 @@ export default function DrugChartWrapper(props) {
     );
     const currentShift = shiftDetailsObj.currentShiftHoursArray;
     const updatedShiftIndex = shiftDetailsObj.shiftIndex;
-    const firstHour = currentShift[0];
-    const lastHour = currentShift[currentShift.length - 1];
-    let startDateTime = getDateTime(new Date(), currentShift[0]);
-    let endDateTime = getDateTime(
-      new Date(),
-      currentShift[currentShift.length - 1] + 1
-    );
+    const [start, end] =
+      shiftDetails.rangeArray[shiftDetails.shiftIndex].split("-");
+    const [startHour, startMinute] = start.split(":");
+    const [endHour, endMinute] = end.split(":");
+    const firstHour = `${startHour}:${startMinute}`;
+    const lastHour = `${endHour}:${endMinute}`;
+    let startDateTime = getDateTime(new Date(), firstHour);
+    let endDateTime = getDateTime(new Date(), lastHour);
     if (lastHour < firstHour) {
       const d = new Date();
       const currentHour = d.getHours();
       if (currentHour > 12) {
         d.setDate(d.getDate() + 1);
-        endDateTime = getDateTime(d, currentShift[currentShift.length - 1] + 1);
+        endDateTime = getDateTime(
+          d,
+          currentShift[currentShift.length - 1].replace(
+            /:\d+$/,
+            `:${endMinute}`
```

Comment:
what are we doing over here?

Reply:
This comes under the handleCurrent function. This is needed to set the minutes part for endTime and startTime, since that won't correct all the time.

eg: If a shift is from 06:00 to 18:30, the shiftArray will have [06:00,07:00,....,18:00]. If we plainly take the last element, it won't have the minute part properly set, so this change is needed